### PR TITLE
Feature/next privous pagination

### PR DESCRIPTION
**Pull request by Nopzen**
# Pagination feature
### Breaking changes:

readEndpoint will no longer resolve a single json response object, but rather a class that contains the response body, and handlers for pagination loading next and loading prev data set.

In stead of getting data like `response.data` its now stored on key `response.body.data` same goes for data set for `include`.
### Benefits

The read endpoint now supports a response with methods to fetch `next` or `prev` data, by calling the `response.loadNext` or `response.loadPrev` the _redux-json-api_ will now handle the call by it self as long as you call the methods.
- [x] Implemented feature
- [x] Local test in dev enviroment
- [ ] Write test case

### CODE REVIEW

**Comment by egeriis** [comment]
Please include a proper description in your PR and also add documentation in the same branch :)

File: src/jsonapi.js [hunk 1]
```diff
@@ -139,7 +139,22 @@ export const readEndpoint = (endpoint, {
...
           onSuccess(json);
-          resolve(json);
+
+          const prevUrl = getPaginationUrl(json, 'prev', apiHost, apiPath);
+          const nextUrl = getPaginationUrl(json, 'next', apiHost, apiPath);
+
+          const getPrevPage = () => dispatch(readEndpoint(prevUrl));
+          const getNextPage = () => dispatch(readEndpoint(nextUrl));
+
+          const resObj = {
```
Comment (egeriis) [inline src/jsonapi.js]: For documentation and clean-code purposes I think it'd be favorable to instantiate a named class to use here.

Comment (Nopzen) [inline src/jsonapi.js]: I would assume insted of putting all this logics into the response, then rather have name class we could call when needed on the .then?

## Example:
```javascript
// inside the client.
import { readEndpoint, setNexturl } from 'redux-json-api';
fetchData() {
  const  { dispatch } = this.props;
  dispatch(readEndpoint('/api/v1/endpoint')
    .then(response => {
       dispatch(setNextUrl(response.data.nextUrl);
     })
}

Comment (egeriis) [inline src/jsonapi.js]: I don't think that's helpful enough to the client. We can do better than that imo.

I was imagining having a class which would be the argument that we're passing through the fulfillment handler. Could look something like this:

```js 
class ApiResponse {
  constructor(prevUrl, nextUrl) {
    this.prevUrl = prevUrl;
    this.nextUrl = nextUrl;
  }

  loadPrev() {
    dispatch(readEndpoint(this.prevUrl));
  }

  loadNext() {
    dispatch(readEndpoint(this.nextUrl));
  }
}
```

**Comment by Nopzen** [comment]
Updated PR description.

**Comment by Nopzen** [comment]
Failing test will be resolved in the 2.0 main branch, these was introduced by merging master in.

**Pull request merged**